### PR TITLE
Add missing Alerters in v2.x

### PR DESCRIPTION
**Pull request by lucasponce**
I am trying to consolidate the features that are not documented (or poorly documented).
More thinking to point people to a doc which is not only based on the javadoc.

### CODE REVIEW

File: src/main/jbake/content/community/docs/developer-guide/alerts-v2.adoc [hunk 1]
```diff
@@ -872,6 +872,253 @@ Url can be a list of valid link:https://hc.apache.org/httpcomponents-core-ga/htt
 | Used for X-Proxy-Remote-User HTTP header.
 |=======================
 
+== Prometheus Integration
+
+=== Prometheus Alerter
+
+The Prometheus Alerter will listen for triggers tagged with "prometheus" tag. the Prometheus evaluations and interacts with the Alerts system. +
```
Comment (jshaughn) [inline src/main/jbake/content/community/docs/developer-guide/alerts-v2.adoc]: I'm not sure what you are trying to say here, it looks maybe like something that you mean to remove?

> the Prometheus evaluations and interacts with the Alerts system.

File: src/main/jbake/content/community/docs/developer-guide/alerts-v2.adoc [hunk 2]
```diff
@@ -872,6 +872,253 @@ Url can be a list of valid link:https://hc.apache.org/httpcomponents-core-ga/htt
...
+ +
+externalcondition.alerterId = "prometheus" +
+externalcondition.expression = <BooleanExpression> | <ALERTSExpression> +
+|=======================
+
+=== Prometheus Alerter Endpoints
+
+|===
+| GET | /hawkular/alerter/status | link:#++GET_P8S_STATUS++[Status endpoint for Prometheus Alerter Integrations]
+| POST | /hawkular/alerter/notifications | link:#++POST_P8S_NOTIFICATIONS++[Listen Prometheus Notifications.]
```
Comment (jshaughn) [inline src/main/jbake/content/community/docs/developer-guide/alerts-v2.adoc]: [Consume Prometheus Alertmanager WebHook Notifications]

File: src/main/jbake/content/community/docs/developer-guide/alerts-v2.adoc [hunk 3]
```diff
@@ -872,6 +872,253 @@ Url can be a list of valid link:https://hc.apache.org/httpcomponents-core-ga/htt
...
+
+==============================================
+
+
+==============================================
+
+[[POST_P8S_NOTIFICATIONS]]
+*Endpoint POST `/awkular/alerter/notifications`*
+
+Receive a Prometheus Notification and store it as a Hawkular Event. Expects a format like: +
```
Comment (jshaughn) [inline src/main/jbake/content/community/docs/developer-guide/alerts-v2.adoc]: Receive a Prometheus Alertmanager WebHook Notification...

File: src/main/jbake/content/community/docs/developer-guide/alerts-v2.adoc [hunk 4]
```diff
@@ -872,6 +872,253 @@ Url can be a list of valid link:https://hc.apache.org/httpcomponents-core-ga/htt
...
+
+|notification.groupLabels.tenant|event.tenantId
+|notification.groupLabels.dataId|event.dataId
+|"prometheus"|event.category
+|notification.commonAnnotations.description|event.text
+|notification.commonAnnotations.summary|event.text
+|notification.groupLabels.alertName|event.text
+|notification.groupLabels.alertName|event.context.alertName
+|notification|event.context.json
+
```
Comment (jshaughn) [inline src/main/jbake/content/community/docs/developer-guide/alerts-v2.adoc]: We may need to be more clear about how the event.text field is set.  It is the first non-empty value of:
1) notification.commonAnnotations.description
2) notification.commonAnnotations.summary
3) notification.groupLabels.alertName
4) "notification"

File: src/main/jbake/content/community/docs/developer-guide/alerts-v2.adoc [hunk 5]
```diff
@@ -872,6 +872,253 @@ Url can be a list of valid link:https://hc.apache.org/httpcomponents-core-ga/htt
...
+
+==============================================
+
+== Apache Kafka Integration
+
+=== Apache Kakfa Alerter
+
+The Kafka Alerter will listen for triggers tagged with "Kafka" tag. The Alerter will create a consumer to a Kafka Topic
+with the config provider into the Trigger context. The Alerter will convert Kafka records into Hawkular Data or Events
```
Comment (jshaughn) [inline src/main/jbake/content/community/docs/developer-guide/alerts-v2.adoc]: Maybe: "config provided in the Trigger context."

File: src/main/jbake/content/community/docs/developer-guide/alerts-v2.adoc [hunk 6]
```diff
@@ -872,6 +872,253 @@ Url can be a list of valid link:https://hc.apache.org/httpcomponents-core-ga/htt
...
+and send them into the Alerting engine.
+
+The Kafka Alerter uses the following conventions for trigger tags and context:
+
+[cols="^2,10"]
+|=======================
+| *Required* |
+_trigger.tags["Kafka"] = "<value reserved for future uses>"_ +
+ +
+ An "Kafka" tag is required for the alerter to detect this trigger will listen to a Kafka topic. +
```
Comment (jshaughn) [inline src/main/jbake/content/community/docs/developer-guide/alerts-v2.adoc]: "The "Kafka" tag is required for the alerter to detect this trigger and listen to the Kafka topic."

File: src/main/jbake/content/community/docs/developer-guide/alerts-v2.adoc [hunk 7]
```diff
@@ -872,6 +872,253 @@ Url can be a list of valid link:https://hc.apache.org/httpcomponents-core-ga/htt
...
+ Value is not necessary, it can be used as a description, it is reserved for future uses. +
```
Comment (jshaughn) [inline src/main/jbake/content/community/docs/developer-guide/alerts-v2.adoc]: future use.

File: src/main/jbake/content/community/docs/developer-guide/alerts-v2.adoc [hunk 8]
```diff
@@ -872,6 +872,253 @@ Url can be a list of valid link:https://hc.apache.org/httpcomponents-core-ga/htt
...
+  +
+i.e. +
+trigger.context["kafka.bootstrap.servers"] = "localhost:9092" +
+trigger.context["kafka.group.id"] = "kafka-trigger-group" +
+ +
+See the link:https://kafka.apache.org/0100/javadoc/index.html?org/apache/kafka/clients/consumer/KafkaConsumer.html[KafkaConsumer reference] for more info +
+| *Required* |
+_trigger.context["topic"] = "<kafka topic to listen>"_ +
+ +
+ It defines the Kafka Topic a trigger will listen. In the current version only a single topic for Trigger is allowed. +
```
Comment (jshaughn) [inline src/main/jbake/content/community/docs/developer-guide/alerts-v2.adoc]: "It defines the Kafka Topic on which the trigger will listen.  The current version allows only a single topic per Trigger."

File: src/main/jbake/content/community/docs/developer-guide/alerts-v2.adoc [hunk 9]
```diff
@@ -872,6 +872,253 @@ Url can be a list of valid link:https://hc.apache.org/httpcomponents-core-ga/htt
...
+[cols="^2,8,^2", options="header"]
+|=======================
+| Property |
+Description |
+Default value
+| *kafka* |
+Used as main prefix for Apache Kafka properties. +
+All *kafka*.<property> properties are passed to KafkaConsumer removing the *kafka* prefix. +
+ +
+Properties can be defined per action based. +
```
Comment (jshaughn) [inline src/main/jbake/content/community/docs/developer-guide/alerts-v2.adoc]: Properties can be defined per action.

**Review by jshaughn — COMMENTED** [review]
A few edits for your consideration.

File: src/main/jbake/content/community/docs/developer-guide/alerts-v2.adoc (hunk 1 above)
Comment (lucasponce) [reply src/main/jbake/content/community/docs/developer-guide/alerts-v2.adoc]: I accidentaly chopped the text.
The description is just a moving from the class javadoc into here.

File: src/main/jbake/content/community/docs/developer-guide/alerts-v2.adoc (hunk 2 above)
Comment (lucasponce) [reply src/main/jbake/content/community/docs/developer-guide/alerts-v2.adoc]: Ok, we can fix it here, but the source code was not documented, so I needed to guess a suitable name.

File: src/main/jbake/content/community/docs/developer-guide/alerts-v2.adoc (hunk 3 above)
Comment (lucasponce) [reply src/main/jbake/content/community/docs/developer-guide/alerts-v2.adoc]: Same comment. We can fix it here but the classes need also some comments from the author too :-)

**Comment by lucasponce** [comment]
Done. I have added also some missing doc related the supported plugins.

**Pull request merged**